mypy/checkmember.py

```diff
@@ -548,6 +551,8 @@ def analyze_class_attribute_access(itype: Instance,
             msg.fail(messages.GENERIC_INSTANCE_VAR_CLASS_ACCESS, context)
         is_classmethod = ((is_decorated and cast(Decorator, node.node).func.is_class)
                           or (isinstance(node.node, FuncBase) and node.node.is_class))
+        if is_classmethod:
+            t = expand_type_by_instance_bases(t, itype)
```

Comment:
I think binding should also happen for variables and normal methods. As you noticed, accessing variables may be bot safe for mutable containers, so it indeed makes sense to give an error, but the inferred type should not contain type variables, they should be either bound to actual values, or replaced with `Any`.